### PR TITLE
feat: Assignment 8 Memory management Initial Commit

### PR DESCRIPTION
**Pull request by Subhasri-V-P**
**Background:**
To understand value and reference types, stack and heap, garbage collection and IDisposable Interface.

**Implementation:**
- Task 1
1. An employee and trainee object is created. The properties of those class and structures can be created, modified and displayed. 
2. ModifyId() method is called to understand the difference between value and reference type
- Task 2
1. Calls `CreateLargeArrayOfIntegers` and `CalculationOnLocalVariables` for noticing the changes in memory.
- Task 3
1. Created and deleted a large number of objects, manually influenced garbage collection, and monitored memory usage.
- Task 4
1. The `Program` class creates an instance of `FileWriter` and writes content to a file named "textFile.txt."
2. Then reads and displays the contents of "textFile.txt."

### CODE REVIEW

File: src/ValueAndReferenceTypes/Trainee.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                case (int)OperationToPerform.Add:
+                    this.AddDetails();
+                    break;
+                case (int)OperationToPerform.Modify:
+                    this.ModifyDetails();
+                    break;
+                case (int)OperationToPerform.Display:
+                    this.DisplayDetails();
+                    break;
+            }
```
Comment (Shaalini-Subrammanian) [inline src/ValueAndReferenceTypes/Trainee.cs:70]: Its Good to have default case
```suggestion
       default:
            break;
            }
```

**Review by Shaalini-Subrammanian — CHANGES_REQUESTED** [review]

File: src/GarbageCollection/Program.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            Console.WriteLine($"After garbage collection: {GC.GetTotalMemory(false)}");
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// CreatesObject() - creates objects of large count
+        /// </summary>
+        public static void CreatesObject()
+        {
+            for (int i = 0; i < 100000; i++)
```
Comment (Shaalini-Subrammanian) [inline src/GarbageCollection/Program.cs:45]:
```suggestion
           while (StudentsList.Count < 100000)
           {
                   StudentsList.Add(new Student() { Id = StudentsList.Count + 1 });
           }
```

File: src/ValueAndReferenceTypes/Employee.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// DisplayDetails() function - displays values stored by this object
+        /// </summary>
+        public void DisplayDetails()
```
Comment (Shaalini-Subrammanian) [inline src/ValueAndReferenceTypes/Employee.cs:77]: Shall we use inheritance and override these methods?

**Review by Shaalini-Subrammanian — CHANGES_REQUESTED** [review]

File: src/ValueAndReferenceTypes/Employee.cs (hunk 3 above)
Comment (Subhasri-V-P) [reply src/ValueAndReferenceTypes/Employee.cs:77]: We can use inheritance if both of them were classes But here one is class and the other one is structure

File: src/GarbageCollection/Program.cs (hunk 2 above)
Comment (Subhasri-V-P) [reply src/GarbageCollection/Program.cs:45]: Loop statement updated

**Review by Shaalini-Subrammanian — APPROVED** [review]

**Pull request merged**